File: code/generate_strategies/IR(pseudocode)/python/python/function_sklearn_mean_squared_error.py
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

class MeanSquaredErrorCalculator:
    def calculate(self, X: str, y: str) -> str:
        X = [float(i) for i in X.split(',')]
        y = [float(i) for i in y.split(',')]
        
        if len(X) != len(y):
            raise ValueError("The number of samples in X and y must be the same.")
        
        model = LinearRegression()
        model.fit([[float(i)] for i in X], y)
        y_pred = model.predict([[float(i)] for i in X])
        
        return str(mean_squared_error(y, y_pred))

File: manual/code/generate_strategies/IR(CoT)/java/java/function_sklearn_mean_squared_error.java
package com.example;

import org.apache.commons.math3.stat.regression.SimpleRegression;

public class MeanSquaredErrorCalculator {
    public String calculate(String X, String y) {
        
        String[] xValues = X.split(",");
        String[] yValues = y.split(",");

        double[] xArray = new double[xValues.length];
        double[] yArray = new double[yValues.length];

        
        if (xValues.length != yValues.length) {
            throw new IllegalArgumentException("The number of samples in X and y must be the same.");
        }

        
        for (int i = 0; i < xValues.length; i++) {
            xArray[i] = Double.parseDouble(xValues[i]);
            yArray[i] = Double.parseDouble(yValues[i]);
        }

        
        SimpleRegression regression = new SimpleRegression();
        for (int i = 0; i < xArray.length; i++) {
            regression.addData(xArray[i], yArray[i]);
        }

        
        double[] yPred = new double[xArray.length];
        for (int i = 0; i < xArray.length; i++) {
            yPred[i] = regression.predict(xArray[i]);
        }

        
        double mse = 0.0;
        for (int i = 0; i < yArray.length; i++) {
            double diff = yArray[i] - yPred[i];
            mse += diff * diff;
        }
        mse /= yArray.length;

        return String.valueOf(mse);
    }
}